### PR TITLE
dev: Update rust dev tools

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

Add rust `nightly` toolchain to the required project build tools.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -46,6 +46,7 @@ default:
 
 # Install necessary tools
 install-tools:
```
Comment (albinsuresh) [inline justfile:48]: Now that we're publishing prebuilt images, I'm even wondering if we should just invoke these `just` setup tasks as part of the image build itself. Even the `setup-integration-test` task as well.

Comment (reubenmiller) [reply justfile:48]: We can add the Rust nightly toolchain to the devcontainer build step, but not the integration test setup as that will install the python virtual environment (which is placed inside the workspace and not in the image itself)

**Review by Bravo555 — APPROVED** [review]

File: justfile (hunk 1 above)
Comment (albinsuresh) [reply justfile:48]: Done in 77b214ff. I haven't removed this task as it'd still be useful for those not using a devcontainer.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by reubenmiller — APPROVED** [review]

**Pull request merged**